dash/edit-mode-design

karolkolodziej opened the pull request:
Improved the edit mode sidebars and popup design.

Waiting for confirmation on how the layout should be styled, which will be in a separate PR.

circleci-mu commented:
### File size comparison
Sizes for compiled+gzipped (bold) and compiled files.
| | master | candidate | difference |
|-------------|-------------:|-------------:|-------------:|
| dashboards/dashboards.js | **42.4 kB**<br>145.6 kB | **42.4 kB**<br>145.8 kB | **34 B**<br>252 B |

circleci-mu commented:
## Visual test results - Differences found
Found **1** diffing sample(s). [Please review the differences.](https://vrevs.highsoft.com/pr/19448/review)

sebastianbochan commented on tools/gulptasks/dashboards/scripts.js:
```diff
@@ -70,7 +70,7 @@ async function dashboardsScripts() {
                 ),
                 namespace: 'Dashboards',
                 output: bundleTargetFolder,
-                version: (argv.release || '0.9.9')
```
Which version should be the default? Should we update it each time?

karolkolodziej replied:
I've added this only to test the designs with the proper icons. 
Finally, it is going to be replaced in the `highcharts-utils` we have dedicated issue in the Assana for that.

pawelfus reviewed (changes requested):
There are a few missing changes (number in brackets reflects number on screenshots):
- Hamburger menu:
  - wrong toggle color (1)
- Edit Cell/Row:
  - wrong handles (2)
  - wrong  background pattern (3)
- Sidebar:
  - wrong input/dropdown backgrounds (4)
  - wrong toggle button colors (5)
  - wrong overlay background (6)
  - increase the Confirm/Cancel buttons corners (radius) (7)
  - replace triangles with arrows (8)
- Adding a new component:
  - The implementation is missing the transition as described in [Asana](https://app.asana.com/0/1203234853779882/1204307015727101/f) 
  - Could we also animate the hiding of the AddComponent sidebar? It vanishes right now


![Zrzut ekranu 2023-07-28 o 11 15 03](https://github.com/highcharts/highcharts/assets/1453926/15028d33-f60c-4775-93e9-41af6d97971b)

![Zrzut ekranu 2023-07-28 o 11 12 40](https://github.com/highcharts/highcharts/assets/1453926/c4002b92-6b22-4d82-be4a-d386df95a46d)
![Zrzut ekranu 2023-07-28 o 11 10 16](https://github.com/highcharts/highcharts/assets/1453926/1f2eb6ad-39f3-4b9f-8317-5acac37d3590)

pawelfus commented on samples/dashboards/demo/minimal/demo.js:
```diff
@@ -136,4 +136,6 @@ Dashboards.board('container', {
     }]
-}, true);
+}, true).then(dashboard => {
+    dashboard.editMode.activate();
```
Cleanup needed ;)

pawelfus commented on gfx/dashboards-icons/resize-handle.svg:
Q: It looks a bit strange 🤔 Do you know by any chance if is it because of the GitHub review tool?

karolkolodziej replied:
Hmm you are right. It looks a bit strange but I have no idea why it might look like this 🤷‍♂️

pawelfus reviewed (changes requested):
Looks great, thank you!

In addition to inline comments:

1. Could you try a white stroke instead of black? I think it will be more visible:
![image](https://github.com/highcharts/highcharts/assets/1453926/fbc686f1-6b2d-4724-ab38-49b92cce786c)

pawelfus reviewed (approved):
Looks good, thanks!

(Approving, but I support Sebastian's suggestions)

pawellysy reviewed (approved):
LGTM

karolkolodziej commented:
@sebastianbochan 
1. Let's leave it as it is, I added a separate task to Assana about `scaleElements` to discuss what to do with it.
2. That is how it was designed, let's leave it.
3. Label was added but there are still some other issues that I reported in Assana. Please resolve this in a separate PR.
4. FIxed.
5. Fixed.

sebastianbochan commented:
I'm not convinced we should leave bugs like in point 1 or 3.

sebastianbochan reviewed (changes requested):
I'm not convinced that we should leave bugs like in point 1 or 3.
@pawelfus @oysteinmoseng
